src/main/java/igym/controller/GymController.java

```
@@ -0,0 +1,32 @@
+package igym.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import igym.entities.*;
+import igym.services.GymService;
+
+@RequestMapping(value = "/api")
+@RestController
+public class GymController {
+
+    @Autowired
+    private GymService service;
+
+    @GetMapping(value = "/gyms", produces = "application/json")
+    public ResponseEntity<List<Gym>> findAllGyms() {
+        List<Gym> gyms = service.findAllGyms();
+
+        if (gyms.isEmpty()) {
+            throw new ResourceNotFoundException("No gyms found");
+        }
```

I don't think that if the result is an empty list of gyms, we need to throw an exception. Do you think we need?

---

Hey Daniel, I think you should throw this exception in the repository, and then propagate it until the controller